### ISSUE TITLE
Error with the latest version of React Native 0.43.0

### ISSUE THREAD

**Issue by diegorodriguesvieira** [aksonov/react-native-router-flux#1761]
### Version
Tell us which versions you are using: 

- react-native-router-flux v3.38.0
- react-native v0.43.0

### Actual behaviour
http://paste.ubuntu.com/24313585/

### Steps to reproduce

1. react-native init AppTest
2. install react-native-router-flux
3. import react-native-router-flux

**Comment by gabrielgomesferraz** [comment]
+1

**Comment by iscarecrow** [comment]
+1

**Comment by vincicat** [comment]
See #1717

**Comment by sporteman** [comment]
+1

**Comment by Kaihuang724** [comment]
+1

**Comment by diegorodriguesvieira** [comment]
I have had to downgrade to get it working again:
```
"react": "16.0.0-alpha.3"
"react-native": "v0.43.0-rc.3"
"react-test-renderer": "16.0.0-alpha.3"
```

**Comment by anuragsimgeker** [comment]
+1

**Comment by georgest** [comment]
+1

**Comment by windwang** [comment]
+1

**Comment by bowekr** [comment]
+1

**Comment by jnbt** [comment]
+1

**Comment by itswaze** [comment]
any solution on this? I want to update to use FlatLists.
Downgrading to 0.43.0-rc.3 didn't work for me.

**Comment by diegorodriguesvieira** [comment]
@itswaze I'm using FlatList...

My package.json
http://paste.ubuntu.com/24321269/

**Comment by itswaze** [comment]
@diegorodriguesvieira That is not working for me... that's strange.
I get this error when I try to use those versions: "Duplicate module name: react-native-vector-icons"

**Comment by mykonwest** [comment]
+1, woof

**Comment by philohelp** [comment]
Same problem here :

> error: bundling: UnableToResolveError: Unable to resolve module `react/lib/ReactComponentWithPureRenderMixin` from `/Users/macbook/test/node_modules/react-native/Libraries/react-native/react-native-implementation.js`: Module does not exist in the module map or in these directories:
>   /Users/macbook/test/node_modules/react-native/node_modules/react/lib
> ,   /Users/macbook/test/node_modules/react/lib

- dreaming to get rid of nav-exprimental and get back to rnrf
- using flatlist & alike everywhere in my app
- not skilled enough to solve that by myself
- deeply depressed

**Comment by philohelp** [comment]
@diegorodriguesvieira I copy pasted you package.json and it works here too.

**Comment by philohelp** [comment]
Same issue on [React-native navigation](https://github.com/react-community/react-navigation/issues/923)

**Comment by beiming** [comment]
Same problem

> error: bundling: UnableToResolveError: 
Unable to resolve module `react/lib/ReactComponentWithPureRenderMixin` from `xxx/node_modules/react-addons-pure-render-mixin/index.js`: 
Module does not exist in the module map or in these directories: xxxx/node_modules/react/lib

**Comment by ShMcK** [comment]
It may be that React 16 alpha.3+ no longer supports mixins. The React team have [hinted at deprecating mixins](https://facebook.github.io/react/blog/2016/07/13/mixins-considered-harmful.html) for a while now.

The pure render mixin could be replaced with a `shouldComponentUpdate` hook that checks for reference and/or deep equality between the old and next props.

According to the [PureRenderMixin docs](https://facebook.github.io/react/docs/pure-render-mixin.html):

> Under the hood, the mixin implements `shouldComponentUpdate`, in which it compares the current props and state with the next ones and returns false if the equalities pass.

The solution may be something as simple as [Dan Abramov's shallowEqual used in react-pure-render](https://github.com/gaearon/react-pure-render/blob/master/src/shallowEqual.js) and replacing the mixin uses with the following:

```js
shouldComponentUpdate() {
  return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
}
```

**Comment by satsuki0221** [comment]
Although it may not be good,
yarn add react-addons-pure-render-mixin
I think it will work.

**Comment by wjnxiang** [comment]
+1

**Comment by sushant6309** [comment]
+1

**Comment by EnriqueDev** [comment]
+1

**Comment by bilby91** [comment]
@satsuki0221 I tried that and it didn't work!

**Comment by satsuki0221** [comment]
This is necessary as well.
react-static-container

I confirmed that it works with these two
yarn add react-static-container react-addons-pure-render-mixin

**Comment by bilby91** [comment]
@satsuki0221 I still have the same problem with:

```
"react-addons-pure-render-mixin": "^15.4.2",
"react-static-container": "^1.0.1",
```

```
➜  app git:(bump-react-native-0.43.1) ✗ react-native bundle --platform ios --dev false --entry-file index.ios.js --bundle-output ios/main.jsbundle
Scanning 764 folders for symlinks in /Users/bilby91/Projects/app/node_modules (12ms)
Scanning 764 folders for symlinks in /Users/bilby91/Projects/app/node_modules (12ms)
Loading dependency graph, done.

Unable to resolve module `react/lib/ReactComponentWithPureRenderMixin` from `/Users/bilby91/Projects/app/node_modules/react-addons-pure-render-mixin/index.js`: Module does not exist in the module map or in these directories:
  /Users/bilby91/Projects/app/node_modules/react/lib
,   /Users/bilby91/node_modules/react/lib

This might be related to https://github.com/facebook/react-native/issues/4968
To resolve try the following:
  1. Clear watchman watches: `watchman watch-del-all`.
  2. Delete the `node_modules` folder: `rm -rf node_modules && npm install`.
  3. Reset packager cache: `rm -fr $TMPDIR/react-*` or `npm start --reset-cache`.
```

**Comment by satsuki0221** [comment]
@bilby91 

confirming my actions, I had done the following procedure

yarn upgrade react-native
yarn upgrade react@16.0.0-alpha.6
yarn add react-static-container react-addons-pure-render-mixin
yarn upgrade

**Comment by bilby91** [comment]
@satsuki0221 The `npm upgrade` did the trick. The problem was with the version of `react-addons-pure-render-mixin`, you need `react-addons-pure-render-mixin@15.5.0` in order to run correctly.

Thanks!

**Comment by bilby91** [comment]
So, temporal solution will be to add this to `package.json`:

```
"react-addons-pure-render-mixin": "^15.5.0",
"react-static-container": "^1.0.1",
```

**Comment by stefan0s** [comment]
it has dependency with "react-addons-pure-render-mixin" which it is removed from React 16.alpha3

**Comment by NightFarmer** [comment]
+1

**Comment by Silvercast** [comment]
+1

**Comment by wuyunqiang** [comment]
how to fix it,me to

**Comment by beiming** [comment]
`npm install --save react-native-router-flux` to fix, just now.

**Comment by stefan0s** [comment]
I don't see any progress, so we will uninstall and move on react-navigation 

https://github.com/react-community/react-navigation

**Comment by jnbt** [comment]
@ShMcK Wouldn't be instrumenting `React.PureComponent` the more "future-ready" solution? https://facebook.github.io/react/docs/react-api.html#react.purecomponent

**Comment by jjdp** [comment]
on rn 0.43.3 it doesnt give me this error anymore

**Comment by ShMcK** [comment]
Definitely.

**Comment by Amurmurmur** [comment]
Still no fix? 
This will require recoding and reimplementation of the navigation.
But well, react-navigation seems like a popular replacement. 👍

**Comment by jjdp** [comment]
@satsuki solution is working fine

**Comment by diegorodriguesvieira** [comment]
Has the problem been solved in version 0.44 of react-native?
I updated and it works ...

My current packages:
```
  "dependencies": {
    "react": "16.0.0-alpha.6",
    "react-native": "0.44.0",
    "react-native-router-flux": "^3.38.0",
  },
  "devDependencies": {
    "babel-jest": "20.0.0",
    "babel-preset-react-native": "1.9.1",
    "jest": "20.0.0",
    "react-test-renderer": "16.0.0-alpha.6"
  },
```
